src/components/FavoritesLink/index.js

```diff
@@ -2,21 +2,48 @@ import { FavoritesSidebarContainer } from "./styled";
 import useAuth from '../../hooks/useAuth';
 import SidebarPanel from "../FavoritesSidebar";
 import useUserLikes from "../../hooks/useUserLikes";
-import { useContext } from "react";
+import { useContext, useEffect, useState } from "react";
 import { GlobalContext } from "../../context/GlobalContext";
+import useUserCollection from "../../hooks/useUserCollection";
+import { onSnapshot } from "firebase/firestore";
+import { getAllUserLinksFromCollection } from "../../firebase/userCollections";
 
 export default function FavoritesPanel() {
     const { user } = useAuth();
+    const [likes, setLikes] = useState([]);
     const { initialLikes } = useUserLikes(user);
     const { isSidebarOpen } = useContext(GlobalContext);
+    const { reference } = useUserCollection('Likes', user);
 
-    return !user || !initialLikes ? null : (
-        <>  
-            <FavoritesSidebarContainer isSidebarOpen={isSidebarOpen}>
-                <SidebarPanel
-                    initialLikes={initialLikes || []} 
-                />
-            </FavoritesSidebarContainer>
-        </>
+    useEffect(() => {
+        if (!likes.length && initialLikes && initialLikes.length) {
+            setLikes(initialLikes)
+        };
+    }, [initialLikes, likes]);
+
+    useEffect(() => {
+        let unsubscribe;
+        if (reference) {
+            unsubscribe = onSnapshot(reference, (snap) => {
+                if (!snap.data().items) return;
+                const userLinks = snap.data().items;
+                getAllUserLinksFromCollection(userLinks).then((links) => {
+                    if (links && links.length) setLikes(links)
+                });
+            });
+        }
+        return () => {
+            if (unsubscribe) unsubscribe();
+        }
+    }, [reference])
```

Comment:
```suggestion
    useEffect(() => {
        if (!reference) {
            return;
        }
        
            return onSnapshot(reference, (snap) => {
                if (!snap.data().items) {
                    return;
                }
                const userLinks = snap.data().items;
                getAllUserLinksFromCollection(userLinks).then((links) => {
                    if (links && links.length) setLikes(links)
                });
            });
    }, [reference])
```

1. return faster and avoid hadouken code https://robsonalves.dev/7-simple-tips-to-write-a-better-code/
2. onSnapshot should (if I don't miss anything) always return the unsubs function or undefined